internal/directives/output_composer.go

```diff
@@ -0,0 +1,70 @@
+package directives
+
+import (
+	"context"
+	"fmt"
+	"maps"
+
+	"github.com/xeipuuv/gojsonschema"
+
+	kargoapi "github.com/akuity/kargo/api/v1alpha1"
+)
+
+func init() {
+	builtins.RegisterPromotionStepRunner(newOutputComposer(), nil)
+}
+
+// outputComposer is an implementation of the PromotionStepRunner interface
+// that allows composing outputs from previous steps into new outputs.
+//
+// It works based on the PromotionStepContext.Config field allowing to an
+// arbitrary number of key-value pairs to be exported as outputs.
+// Because the values are allowed to be expressions and can contain
+// references to outputs from previous steps, this allows for remapping
+// the outputs of previous steps to new keys, or even combining them
+// into new structures.
```

Comment:
I know docs are coming in a follow-up. In the interim, I think a small example at the tail end of this comment would be helpful.